src/main/java/org/opentripplanner/model/UpdateSuccess.java

```diff
@@ -0,0 +1,25 @@
+package org.opentripplanner.model;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import org.opentripplanner.framework.collection.ListUtils;
+
+public record UpdateSuccess(List<WarningType> warnings) {
+  public static UpdateSuccess noWarnings() {
+    return new UpdateSuccess(List.of());
+  }
+
+  public static UpdateSuccess ofWarnings(WarningType... warnings) {
+    return new UpdateSuccess(Arrays.asList(warnings));
+  }
+
+  public UpdateSuccess addWarnings(Collection<WarningType> addedWarnings) {
+    return new UpdateSuccess(ListUtils.combine(this.warnings, addedWarnings));
+  }
+
+  public enum WarningType {
+    UNKNOWN_STOPS_REMOVED_FROM_ADDED_TRIP,
+    NOT_MONITORED,
+  }
```

Comment:
No need to change this, but there is a miss-match between the generic class name indicating that this a common "framework" class and the list of waring types. This can become the "hub" in a "hub-spoke" - then having something in it that changes every time a new WARNING type occurs will potentially introduce changes/bugs in all places it is used; Hence a maintenance issue. 

~~Should you use a set instead of list for the warning here, there is no context information so knowing how many warnings there are might (not) be useful?~~

JavaDoc might help here, it toke me a while to find out what this was used for. Tip: When creating an issue report/monitoring framework it is usually better to let the clients define their own warning types, not using an enum. Enums are great if you need to programatically take an action based on it... We should also put this class and potentially others into a framework package (if I understand it right, this is used to log error/warnings metrics about the updaters, maybe `o.o.framework.metrics.updater`). 

This is also missing JavaDoc on all public members - class, method and enum values. 

Minor: ofWarnings(A, B) != ofWarnings(B, A)

Reply:
You're right. It was in a framework package but it's actually specific a realtime updates not general. I've moved it into the update package.

You're also right about the Javadoc. I added it.